File: Itinere_3/Laravel/department_employee/app/Http/Controllers/DepartmentController.php
<?php

namespace App\Http\Controllers;

use App\Models\Department;
use Illuminate\Http\Request;

class DepartmentController extends Controller
{
    public function index()
    {
        return view("departments.index", ["departments" => Department::all()]);
    }

    public function create()
    {
        return view("departments.create");
    }

    public function store(Request $request)
    {
        Department::create($request->all());
        return redirect()->route("departments.index");
    }

    public function show(Department $department)
    {
        return view("departments.show", ["d" => $department]);
    }

    public function edit(Department $department)
    {
        return view("departments.edit", ["d" => $department]);
    }

    public function update(Request $request, Department $department)
    {
        $department->update($request->all());
        return redirect()->route("departments.index");
    }

    public function destroy(Department $department)
    {
        $department->delete();
        return redirect()->route("departments.index");
    }
}

File: Itinere_3/Spring-Boot/department_employee/src/main/java/com/example/department_employee/controllers/DepartmentController.java
package com.example.department_employee.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.*;

import com.example.department_employee.Repositories.DepartmentReposity;
import com.example.department_employee.models.Department;

@Controller
public class DepartmentController {
    private final DepartmentReposity repo;

    public DepartmentController(DepartmentReposity repo){
        this.repo = repo;
    }

    @GetMapping("/")
    String welcome(){
        return "/welcome";
    }

    @GetMapping("/departments")
    String index(Model model){
        model.addAttribute("departments", repo.findAll());
        return "/departments/index";
    }

    @GetMapping("/departments/create")
    String create(){
        return "/departments/create";
    }

    @PostMapping("/departments/store")
    String store(Department department){
        repo.save(department);
        return "redirect:/departments";
    }

    @GetMapping("/departments/show")
    String show(Model model, Long id){
        model.addAttribute("d", repo.findById(id).orElse(null));
        return "/departments/show";
    }

    @GetMapping("/departments/edit")
    String edit(Model model, Long id){
        model.addAttribute("d", repo.findById(id).orElse(null));
        return "/departments/edit";
    }

    @PostMapping("/departments/update")
    String update(Department department){
        repo.save(department);
        return "redirect:/departments";
    }

    @PostMapping("/departments/destroy")
    String destroy(Department department){
        repo.deleteById(department.getId());
        return "redirect:/departments";
    }
}
